spec/models/request_spec.rb

```diff
@@ -268,6 +268,22 @@
         expect(User.find_by(email: 'jstanford@stanford.edu').name).to eq 'Jane Stanford'
       end
 
+      it 'should not duplicate user records when both library ID and email is provided' do
+        expect(User.where(library_id: '12345', email: 'jstanford@stanford.edu').length).to eq 0
+        User.create(library_id: '12345', email: 'jstanford@stanford.edu')
+        expect(User.where(library_id: '12345', email: 'jstanford@stanford.edu').length).to eq 1
+        Request.create!(
+          item_id: '1234',
+          origin: 'GREEN',
+          origin_location: 'STACKS',
+          user_attributes: {
+            library_id: '12345',
+            email: 'jstanford@stanford.edu'
+          }
+        )
+        expect(User.where(library_id: '12345', email: 'jstanford@stanford.edu').length).to eq 1
+      end
```

Comment:
Hmm, I'm curious if this actually tests our bug.  Does this fail before your change?